### PR TITLE
haxe: fixed --HEAD

### PR DESCRIPTION
**Pull request by andyli**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Updated the haxe formula to adapt recent changes in upstream.
The usage of opam is borrowed from other ocaml-based formulae, e.g. https://github.com/Homebrew/homebrew-core/blob/master/Formula/coccinelle.rb

### CODE REVIEW

**Comment by kevinresol** [comment]
Any updates on this? 
Currently homebrew fails to install haxe --HEAD

**Comment by andyli** [comment]
What's holding this PR back?

File: Formula/haxe.rb [hunk 1]
```diff
@@ -15,16 +15,24 @@ class Haxe < Formula
 
   depends_on "ocaml" => :build
   depends_on "camlp4" => :build
+  depends_on "opam" => :build if build.head?
```
Comment (ilovezfs) [inline Formula/haxe.rb]: This should be moved into a `head do` block

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
otherwise looks good
```
==> Summary
🍺  /usr/local/Cellar/haxe/HEAD-c8e43d3: 2,333 files, 19MB, built in 1 minute 2 seconds
iMac-TMP:~ joe$ brew test haxe
Testing haxe
==> Using the sandbox
==> /usr/local/Cellar/haxe/HEAD-c8e43d3/bin/haxe -v Std
==> /usr/local/Cellar/haxe/HEAD-c8e43d3/bin/haxelib version
```

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks @andyli :rocket:

**Pull request merged**